components/public-api/gitpod/v1/envvar.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service EnvironmentVariableService {
+  // ListEnvironmentVariables returns all environment variables for the
+  // authenticated user.
+  //
+  // +return NOT_FOUND User does not have access to a workspace with the given
+  // ID +return NOT_FOUND Workspace does not exist
+  rpc ListEnvironmentVariables(ListEnvironmentVariablesRequest)
+      returns (ListEnvironmentVariablesResponse) {}
+
+  // UpdateEnvironmentVariable updates an environment variable for the
+  // authenticated user.
+  rpc UpdateEnvironmentVariable(UpdateEnvironmentVariableRequest)
+      returns (UpdateEnvironmentVariableResponse) {}
+
+  // CreateEnvironmentVariable creates a new environment variable for the
+  // authenticated user.
+  rpc CreateEnvironmentVariable(CreateEnvironmentVariableRequest)
+      returns (CreateEnvironmentVariableResponse) {}
+
+  // DeleteEnvironmentVariable deletes an environment variable for the
+  // authenticated user.
+  rpc DeleteEnvironmentVariable(DeleteEnvironmentVariableRequest)
+      returns (DeleteEnvironmentVariableResponse) {}
+}
+
+message UserScope {
+  string repository_pattern = 1;
+}
+
+message ConfigurationScope {
+  string configuration_id = 1;
+  bool censored = 2;
+}
+
+message EnvironmentVariable {
+  string id = 1;
+  string name = 2;
+  string value = 3;
+  oneof scope {
+    UserScope user_scope = 4;
+    ConfigurationScope configuration_scope = 5;
+  }
+}
+
+message ListEnvironmentVariablesRequest {
+  string workspace_id = 1;
+  string configuration_id = 2;
+}
+
+message ListEnvironmentVariablesResponse {
+  repeated EnvironmentVariable environment_variables = 1;
+}
+
+message UpdateEnvironmentVariableRequest {
+  optional string env_var_name = 1;
+  optional string env_var_value = 2;
+  oneof scope {
+    UserScope user_scope = 3;
+    ConfigurationScope configuration_scope = 4;
```

Comment:
it breaks partial pattern here,

we should have then 
```
optional bool censored = 3; 
```

it would be a case for nested messages to override global, i..e
```proto
message UpdateEnvironmentVariableRequest {
  string env_var_id = 1; // required
  optional string name = 2;
  optional string value = 3;
  message UserScope {
    optional string repository_pattern = 1;
  }
  message ConfigurationScope {
    // configuration_id should not be here it is not updatable
    optional bool censored = 1;
  }
  oneof scope {
    optional UserScope userScope = 4;
    optional UserScope configurationScope = 5;
  }
}
```

Reply:
it's needed to actually call `listProjectEnvVars` internally, thoughts?